In [45]:
import pandas as pd 
import numpy as np 
import matplotlib
import seaborn

In [46]:
inven_path='Data\\Inventory Dataset.xlsx'
new_inv_path='Data\\New Inventory.xlsx'

In [47]:
df_inventory=pd.read_excel(inven_path)
df_new_arrivals=pd.read_excel(new_inv_path)

## Inventory Dataset.xlsx

In [48]:
print("The Inventory dataset contains {} rows and {} columns.".format(df_inventory.shape[0],df_inventory.shape[1]))


The Inventory dataset contains 356 rows and 3 columns.


In [49]:
len(df_inventory['Item Number'].unique())

356

## New Inventory.xlsx

In [50]:
print("The New inventory dataset contains {} rows and {} columns.".format(df_new_arrivals.shape[0],df_new_arrivals.shape[1]))

The New inventory dataset contains 1039 rows and 3 columns.


In [51]:
len(df_new_arrivals['Item Number'].unique())

213

In [52]:
len(df_new_arrivals['Inventory Receive'].unique())

23

The inventory data set contains 356 unique item numbers and the number of items in that inventory. The new inventory dataset contains 1,039 rows of numbers <br> 
showing how many new items were added to inventory and when they arrived. The new inventory dataset contains 213 unique items, which means that out of 356 items,<br>
only 213 items were received on 23 different unique dates.

In [53]:
df_new_arrivals['Inventory Receive']=df_new_arrivals['Inventory Receive'].dt.strftime('%d/%m/%Y')
df_inventory['Date']=df_inventory['Date'].dt.strftime('%d/%m/%Y')
#df_inventory=df_inventory.sort_values(by=['Inventory Receive'],)

## Merging both files

In [54]:
df_merged = pd.merge(df_inventory, df_new_arrivals, on='Item Number', how='outer')
df_merged=df_merged.sort_values(by=['Inventory Receive'],ascending=False)

In [55]:
#df_merged['Inventory Receive']=df_merged['Inventory Receive'].dt.strftime('%d/%m/%Y')
#df_merged['Date']=df_merged['Date'].dt.strftime('%d/%m/%Y')

In [56]:
pivot_table = df_merged.pivot_table(values='Total Item Qty', 
                             index='Item Number', 
                             columns='Inventory Receive', 
                             aggfunc='sum', 
                             fill_value=0)


In [57]:

for item_num,row in pivot_table.iterrows():
    try:
        on_stock = df_inventory[df_inventory['Item Number'] == item_num]['Inventory']
        on_stock=int(on_stock)
    except:
        on_stock=0

    for k in pivot_table.columns:
        on_stock=on_stock+row[k]
        row[k]=on_stock
        
    

C:\Users\prabh\AppData\Local\Temp\ipykernel_17500\4281683759.py:4: FutureWarning: Calling int on a single element Series is deprecated and will raise a TypeError in the future. Use int(ser.iloc[0]) instead
  on_stock=int(on_stock)


In [59]:
df_new_arrivals['Inventory Receive'].unique()

array(['07/07/2023', '09/07/2023', '21/07/2023', '16/08/2023',
       '13/10/2023', '10/07/2023', '04/08/2023', '08/09/2023',
       '25/08/2023', '29/09/2023', '24/09/2023', '14/07/2023',
       '01/12/2023', '22/08/2023', '28/07/2023', '18/08/2023',
       '08/10/2023', '22/10/2023', '15/12/2023', '08/12/2023',
       '26/11/2023', '15/09/2023', '22/09/2023'], dtype=object)

In [58]:
df_new_arrivals.groupby('Item Number').sum()

Inventory Receive  Total Item Qty
Item Number                                                                   
064-01-0003                               13/10/202329/09/2023            7352
064-01-0005                               13/10/202329/09/2023            9900
064-01-0012                               13/10/202329/09/2023           10084
064-01-0013                               13/10/202329/09/2023            8048
064-01-0016                               13/10/202329/09/2023            7636
...                                                        ...             ...
064-18-9182  16/08/202313/10/202329/09/202309/07/202304/08/...           34884
064-18-9667                               13/10/202329/09/2023            1984
064-18-9724  21/07/202313/10/202329/09/202309/07/202304/08/...           11316
064-18-9904                               01/12/202315/12/2023           23800
064-18-9938  16/08/202313/10/202329/09/202309/07/202304/08/...           15448

[213 rows x 2 columns]